src/linux/component.rs

```diff
@@ -1,160 +1,407 @@
+///   see below for special cases
+/// - Optionnal: max value defined in `tempN_max`
+/// - Optionnal: critical value defined in `tempN_crit`
+///
+/// Where `N` is a u32 associated to a sensor like `temp1_max`, `temp1_input`.
+///
+/// ## Special case: Disk drive with `drivetemp` module
+///
+/// For some crazy rational, hardware monitoring exposed by `drivetemp` has no label so the `label of a `Component`
+/// deduced.
+///
+/// So the `device/model` content will be used instead of `tempN_label` content.
+///
+/// ## Special case: RasperryPy CPU and some vendor sensors
+///
+/// Have empty label, file exist but is empty, they will be labelled `Components N`.
+///
+/// ## Doc to Linux kernel API.
+///
+/// Kernel hwmon API: https://www.kernel.org/doc/html/latest/hwmon/hwmon-kernel-api.html
+/// DriveTemp kernel API: https://docs.kernel.org/gpu/amdgpu/thermal.html#hwmon-interfaces
+/// Amdgpu hwmon interface: https://www.kernel.org/doc/html/latest/hwmon/drivetemp.html
+fn append_files(components: &mut Vec<Component>, folder: &Path) -> Option<()> {
+    let dir = read_dir(folder).ok()?;
+    let mut matchings: HashMap<u32, Component> = HashMap::with_capacity(10);
+    for entry in dir.flatten() {
+        let entry = entry.path();
+        let filename = entry.file_name().and_then(|x| x.to_str()).unwrap_or("");
+        if entry.is_dir() || !filename.starts_with("temp") {
+            continue;
+        }
+
+        let (id, item) = filename.split_once('_')?;
+        let id = id.get(4..)?.parse::<u32>().ok()?;
+
+        let component = matchings.entry(id).or_insert_with(|| Component::default());
+        let name = get_file_line(&folder.join("name"), 16);
+        component.name = name.unwrap_or(String::new());
+        let device_model = get_file_line(&folder.join("device/model"), 16);
+        component.device = device_model.unwrap_or(String::new());
+        fill_component(component, id, item, folder, filename);
+    }
+    let compo = matchings
+        .into_iter()
+        .map(|(k, mut c)| {
+            // sysinfo expose a generic interface with a `label`.
+            // Problem: lot of sensor don't have a label or a device model! ¯\_(ツ)_/¯
+            // So let's pretend we have a unique label!
+            //
+            // All sensor have a `name` on linux hwmon it's mandatory.
+            // - if a sensor has no label: check if don't have a model name, if so it's use it.
+            // - if not do something from id and name at best.
+            let has_device = !c.device.is_empty();
+            let has_label = !c.label.is_empty();
+            c.label = match (has_label, has_device) {
+                (true, true) => format!("{} {} {}", c.name, c.label, c.device),
+                (true, false) => format!("{} {}", c.name, c.label),
+                (false, true) => format!("{} {}", c.name, c.device),
+                (false, false) => format!("{} {}", c.name, k),
+            };
```

Comment:
@GuillaumeGomez @linkmauve did something, since we don't group (yet) like `sensors` we may report with always name and other data if possible